vertical-pod-autoscaler/pkg/recommender/main.go

```diff
@@ -307,3 +321,11 @@ func run(healthCheck *metrics.HealthCheck, commonFlag *common.CommonFlags) {
 		healthCheck.UpdateLastActivity()
 	}
 }
+
+func setGlobalMaxAllowed(globalMaxAllowed *apiv1.ResourceList, key apiv1.ResourceName, value resource.Quantity) {
```

Comment:
What are your thoughts on converting this function into an "initializer" for the globalMaxAllowed object?

In `main()` you would do:

`globalMaxAllowed := initGlobalMaxAllowed()`

And then in this function:

```
func initGlobalMaxAllowed() apiv1.ResourceList {
  result := make(apiv1.ResourceList)
  if !maxAllowedCPU.Quantity.IsZero() {
    result[apiv1.ResourceCPU] = maxAllowedCPU.Quantity
  }
  ... same for memory ...
  return result
}
```

This way you avoid playing around with pointers and the function is fully self contained.